sql/src/main/java/io/crate/metadata/Functions.java

```diff
@@ -98,9 +101,15 @@ public FunctionImplementation getBuiltin(String name, List<DataType> argumentsTy
      * @return a function implementation.
      * @throws UnsupportedOperationException if no implementation is found.
      */
-    public FunctionImplementation getUserDefined(String name, List<DataType> argumentsTypes) throws UnsupportedOperationException {
-        FunctionResolver dynamicResolver = udfFunctionResolvers.get().get(name);
-        FunctionImplementation impl = resolveFunctionForArgumentTypes(argumentsTypes, dynamicResolver);
+    public FunctionImplementation getUserDefined(String schema, String name, List<DataType> argumentsTypes) throws UnsupportedOperationException {
+        if (schema == null) {
+            throw createUnknownFunctionException(name, argumentsTypes);
```

Comment:
i would move this check down to `getQualified`. `getUserDefined` shouldn't be called with a `null` schema.